components/brave_rewards/resources/tip/lib/theme_loader.ts

```diff
@@ -0,0 +1,17 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import defaultTheme from 'brave-ui/theme/brave-default'
+
+export function injectThemeVariables (element: HTMLElement) {
+  for (const [key, value] of Object.entries(defaultTheme.color)) {
+    element.style.setProperty(`--brave-color-${key}`, String(value))
+  }
```

Comment:
This is great.

1) Could you put it in `components/common/` so that we can use it for other UI. I for certain would.

2) I'm not sure that we should be using `defaultTheme` directly. We should be ascertaining what the current theme is based on whether we're in light or dark mode, and then switching the theme (and replacing the variable values) when the user selected theme changes. We used to have elaborate listeners, but I think these days we can use `window.matchMedia('(prefers-color-scheme: dark)')` and put a listener on that. The idea is that you can either use the default dark and light themes, or subclass them and make specific overrides for each for your specific UI, if applicable.

Reply:
Moving this out so that other UIs can use it was part of the plan (hope?), but I think we should do it as a follow-on, so that we'll have a bit more time to think about how to generalize it properly. We should be able to easily migrate the code in here to use a helper that we promote to "common".

Really interesting point about `matchMedia`!